lib/experimental/inject_parser.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Parses a snippet into multiple file types separated by
+/// `{% begin <filename> %}` and `{% end <filename> %}` markers
```

Comment:
How did you arrive at these particular tokens? Would it be possible to use some other tag inside the code element (or multiple code elements) to delineate the individual files?

Reply:
I'm mostly thinking of this in terms of "I can't remember the last time I successfully maintained a regex-based parser written by someone else." 😄

Reply:
It's Jekyll syntax but now that you mention it, this could be a bad idea; I'll test this with a Jekyll app to make sure it works.

re: regex parser, there's tests to make it easier but I agree they difficult to maintain. I'm open to other ideas to make it better.

Reply:
Not surprisingly, `{% begin <filename> %}` and `{% end <filename> %}` syntax isn't compatible with Jekyll. I'll see if I we use a syntax that other static site generators aren't using.

Reply:
I’d investigate using a real parser, eg https://pub.dev/packages/petitparser

Reply:
I'll see what the flutter docs are doing. They have a nice syntax for samples:

````
{@tool sample}
```dart
AnimatedIcon(
  icon: AnimatedIcons.menu_arrow,
  progress: controller,
  semanticLabel: 'Show menu',
)
```
{@end-tool}
````

Reply:
FWIW [dartddoc uses a regex](https://github.com/dart-lang/dartdoc/blob/master/lib/src/model.dart#L124)